### ISSUE TITLE
train one class model--there is a wired drop with  performance

### ISSUE THREAD

**Issue by GuitarYourself** [ultralytics/yolov5#3952]
## ❔Question
I'm training a one class model with 1M images, there is always a drop around the 4-10 epoch. I can not figure what's wrong with it.

## Additional context
![image](https://user-images.githubusercontent.com/4207125/125154837-a97bca80-e18e-11eb-8d89-799c11d9f211.png)

lr0: 0.01  # initial learning rate (SGD=1E-2, Adam=1E-3)
lrf: 0.01  # final OneCycleLR learning rate (lr0 * lrf)
momentum: 0.937  # SGD momentum/Adam beta1
weight_decay: 0.0005  # optimizer weight decay 5e-4
warmup_epochs: 5.0  # warmup epochs (fractions ok)
warmup_momentum: 0.8  # warmup initial momentum
warmup_bias_lr: 0.1  # warmup initial bias lr
box: 0.05  # box loss gain
cls: 0.5  # cls loss gain
cls_pw: 1.0  # cls BCELoss positive_weight
obj: 1.0  # obj loss gain (scale with pixels)
obj_pw: 1.0  # obj BCELoss positive_weight
iou_t: 0.20  # IoU training threshold
anchor_t: 4.0  # anchor-multiple threshold
# anchors: 3  # anchors per output layer (0 to ignore)
fl_gamma: 0.0  # focal loss gamma (efficientDet default gamma=1.5)
hsv_h: 0.015  # image HSV-Hue augmentation (fraction)
hsv_s: 0.7  # image HSV-Saturation augmentation (fraction)
hsv_v: 0.4  # image HSV-Value augmentation (fraction)
degrees: 10.0  # image rotation (+/- deg)
translate: 0.1  # image translation (+/- fraction)
scale: 0.5  # image scale (+/- gain)
shear: 10.0  # image shear (+/- deg)
perspective: 0.0001  # image perspective (+/- fraction), range 0-0.001
flipud: 0.35  # image flip up-down (probability)
fliplr: 0.35  # image flip left-right (probability)
mosaic: 0.2  # image mosaic (probability)#ori: 1.o
mixup: 0.0  # image mixup (probability)

**Comment by glenn-jocher** [comment]
@GuitarYourself you can modify the warmup hyperparameters as you see fit for your custom dataset.

**Comment by GuitarYourself** [comment]
> @GuitarYourself you can modify the warmup hyperparameters as you see fit for your custom dataset.

that's the result after i make no warmup, i directly change the code to make it:
nw = max(round(hyp['warmup_epochs'] * nb), 1000)  # number of warmup iterations, max(3 epochs, 1k iterations)
#nw = min(nw, (epochs - start_epoch) / 2 * nb)  # limit warmup to < 1/2 of training
nw = 0  #make no warmup

**Comment by glenn-jocher** [comment]
@GuitarYourself changing the code is not recommended unless you know what you are doing, but if you're not happy with your results you are free to modify your warmup hyperparameters as you see fit.

**Comment by GuitarYourself** [comment]
> @GuitarYourself changing the code is not recommended unless you know what you are doing, but if you're not happy with your results you are free to modify your warmup hyperparameters as you see fit.

so, you think the sudden drop is caused by the  warmup learning rate?

**Comment by glenn-jocher** [comment]
@GuitarYourself in general warmup is used to prevent instabilities in early training.
